spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -1062,25 +1053,23 @@ expr_e        : CAST type_param_begin qtype type_param_end '(' expr ')'   { $$ =
               | TYPEINFO '(' base_type ')'       { $$ = builder->expressionTypeInfo(builder->expressionType(builder->qualifiedType(std::move($3), hilti::Constness::Mutable)), __loc__); }
-              | expr_f                           { $$ = std::move($1); }
-
-expr_f        : ctor                             { $$ = builder->expressionCtor(std::move($1), __loc__); }
+              | ctor                             { $$ = builder->expressionCtor(std::move($1), __loc__); }
               | ctor_expr                        { $$ = std::move($1); }
-              | '-' expr_f                       { $$ = builder->expressionUnresolvedOperator(hilti::operator_::Kind::SignNeg, {std::move($2)}, __loc__); }
-              | '[' expr FOR local_id IN expr ']'
-                                                 { $$ = builder->expressionListComprehension(std::move($6), std::move($2), std::move($4), {},  __loc__); }
-              | '[' expr FOR local_id IN expr IF expr ']'
-                                                 { $$ = builder->expressionListComprehension(std::move($6), std::move($2), std::move($4), std::move($8),  __loc__); }
-              | expr_g                           { $$ = std::move($1); }
-
-expr_g        : '(' expr ')'                     { $$ = builder->expressionGrouping(std::move($2)); }
+              | '[' expr FOR local_id IN expr ']'{ $$ = builder->expressionListComprehension(std::move($6), std::move($2), std::move($4), {},  __loc__); }
+              | '[' expr FOR local_id IN expr IF expr ']'   { $$ = builder->expressionListComprehension(std::move($6), std::move($2), std::move($4), std::move($8),  __loc__); }
+              | '(' expr ')'                     { $$ = builder->expressionGrouping(std::move($2)); }
               | scoped_id                        { $$ = builder->expressionName(std::move($1), __loc__); }
               | DOLLARDOLLAR                     { $$ = builder->expressionName(std::move("__dd"), __loc__);}
               | DOLLAR_NUMBER                    { // $N -> $@[N] (with $@ being available internally only, not exposed to users)
                                                    auto captures = builder->expressionKeyword(hilti::expression::keyword::Kind::Captures, builder->qualifiedType(builder->typeName("~hilti::Captures"), hilti::Constness::Mutable), __loc__);
                                                    auto index = builder->expressionCtor(builder->ctorUnsignedInteger($1, 64, __loc__), __loc__);
                                                    $$ = builder->expressionUnresolvedOperator(hilti::operator_::Kind::Index, {std::move(captures), std::move(index)}, __loc__);
                                                  }
+    ;
+
+call_expr     : scoped_id '(' opt_exprs ')'      { $$ = builder->expressionUnresolvedOperator(hilti::operator_::Kind::Call, {builder->expressionName(std::move($1)), builder->expressionCtor(builder->ctorTuple(std::move($3), __loc__))}, __loc__); }
```

Comment:
Is it necessary to limit the callee to an ID? I don't think we're using anything else there right now, but in principle we could.

Reply:
If we ever need it, it's most likely not just a specific syntactic construct to add, but actually full expressions: I'm picturing overloading expressions of certain types with a call operator, which our infrastructure would support just fine. With that in mind, I'd prefer to keep `expr` for the callee even with the additional conflicts just so that we know we could go that route still---unless there's something we know is not going to work in the new version (we had plenty conflicts already, but it doesn't seem to hurt anywhere so far). 

Same applies to the new HILTI grammar then.